### PR TITLE
manim: add linked indirect dependencies

### PR DESCRIPTION
**Pull request by bevanjkay**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Seen in: https://github.com/Homebrew/homebrew-core/actions/runs/9094160676?pr=171740

### CODE REVIEW

**Review by carlocab — COMMENTED** [review]
There are still remaining indirect deps.

On Linux:
```
Indirect dependencies with linkage:
  fontconfig
  gcc
  glib
  jpeg-turbo
  libtiff
  openblas
  pango
  zlib
```
On Intel macOS:
```
Indirect dependencies with linkage:
  libxcb
  pango
```
I'd add these to all macOS, though, since they're already in the dep tree anyway.

**Comment by bevanjkay** [comment]
The other ones showed extra Linux deps on the rebuild run too. I'll update it in a couple of hours.

**Comment by bevanjkay** [comment]
@carlocab I have added the extras, put `gcc` as a dep for both because it showed for MacOS and Linux, so it is more likely an actual dependent?

**Comment by carlocab** [comment]
Actually, the `resource`s are a mess.

There are a bunch that duplicate formulae that are already declared as dependencies (e.g. `scipy` on ARM).

We should also just use the `pillow` formula instead of the `resource`, which I suspect will let you get rid of a bunch of these dependencies.

**Comment by p-linnane** [comment]
> Actually, the `resource`s are a mess.
> 
`moderngl-window` doesn't have an sdist on PyPI, so we can't use `bump-python-resources`. I figure that's why this is so messy.

**Comment by carlocab** [comment]
> `moderngl-window` doesn't have an sdist on PyPI, so we can't use `bump-python-resources`. I figure that's why this is so messy.

That might be another reason, but the problems about duplicating formulae dependencies aren't really fixed by `bump-python-resources`, so it would still be messy even if we could use it...

**Comment by carlocab** [comment]
Tried fixing it up a bit. Hopefully it works. 🤞

File: Formula/m/manim.rb [hunk 1]
```diff
@@ -23,17 +23,30 @@ class Manim < Formula
...
+  end
+
   on_linux do
     depends_on "cmake" => :build
     depends_on "patchelf" => :build
   end
 
   on_arm do
-    depends_on "pango"
     depends_on "scipy"
```
Comment (carlocab) [inline Formula/m/manim.rb:50]: As expected, this seems to be a dependency everywhere and not just on ARM.

File: Formula/m/manim.rb [hunk 2]
```diff
@@ -23,20 +23,26 @@ class Manim < Formula
...
   depends_on "python@3.12"
+  depends_on "scipy"
+
+  on_macos do
+    depends_on "gettext"
+    depends_on "harfbuzz"
+  end
 
   on_linux do
     depends_on "cmake" => :build
```
Comment (carlocab) [inline Formula/m/manim.rb:42]: Needing `cmake` here is kinda suspicious, ngl

**Review by p-linnane — APPROVED** [review]

**Comment by bevanjkay** [comment]
Thanks for finishing this up @carload

**Comment by chenrui333** [comment]
some followup PR, https://github.com/Homebrew/homebrew-core/pull/177164

**Pull request merged**